pkg/tasks/delete_repository_snapshots.go

```diff
@@ -52,29 +52,37 @@ func DeleteSnapshotHandler(ctx context.Context, task *models.TaskInfo, _ *queue.
 }
 
 func (d *DeleteRepositorySnapshots) Run() error {
-	snaps, _ := d.fetchSnapshots()
-	if len(snaps) == 0 {
-		return nil
-	}
-	for _, snap := range snaps {
-		_, err := d.deleteRpmDistribution(snap)
-		if err != nil {
-			return err
+	var err error
+	if config.PulpConfigured() {
+		snaps, _ := d.fetchSnapshots()
+		if len(snaps) == 0 {
+			return nil
```

Comment:
If I delete a repo config that doesn't have snapshots enabled, the repo config is only soft deleted. I think because this line here returns when there are no snapshots.

Reply:
good catch!

Reply:
i'll add a test too.  Ohh, and i guess since this is deleting non-snapshotted repos, should i rename the task?

Reply:
and actually with that 'return nil' if you created a repo pointed at a bad url (so no snapshots ever got created) and then deleted it, it'd leave the repo & remote in pulp